### ISSUE TITLE
Stutter service name if user have `Service` suffix

### ISSUE THREAD

**Issue by shuLhan** [micro/protoc-gen-micro#6]
Since not all service can be described by "verb", e.g. "Search", user pick the name style used by protobuf documentation [1][2]. Commit 50515c29 rename suffix of generated type from XxxClient` to `XxxService`. If user also use `Service` as the suffix, the generated output will print `XxxServiceService`.

Please revert that commit, thank you.

[1] https://developers.google.com/protocol-buffers/docs/proto#services
[2] https://developers.google.com/protocol-buffers/docs/style

**Comment by asim** [comment]
Unfortunately the change will not be reverted. Our generated types were producing a conflict with gRPC generated types and we needed to be able to interoperate. We welcome alternative suggestions.

**Comment by asim** [comment]
We can potentially introduce a fix which looks for names with Service suffix and avoid generating ServiceService but this will cause conflicts where you have `Search` and `SearchService` as two services in the same place.

**Comment by shuLhan** [comment]
> We can potentially introduce a fix which looks for names with Service suffix and avoid generating ServiceService but this will cause conflicts where you have Search and SearchService as two services in the same place.

I think that possibility is near 0. In our work scenario, all protobuf related are grouped into one repository where other project can reference to it using import. Assume that one define `Search` and `SearchService` as services, and protoc-gen-micro generate it as `SearchService` and `SearchService`, `protoc` will report an error,

```
test.proto:9:9: "package.SearchService" is already defined in file "search.proto".
```
Do you accept PR for that?

**Comment by asim** [comment]
Yes I'll accept a PR for this. We need to make it clear in a section in the readme with an example so that people are aware of this.
